### ISSUE TITLE
`msgpack` formula generates invalid pkg-config file

### ISSUE THREAD

**Issue by fumoboy007** [Homebrew/homebrew-core#175453]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```console
$ brew config
HOMEBREW_VERSION: 4.3.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: e8430b25a1d1321f32e5093d62b57b5cb7cfb3c3
Last commit: 8 days ago
Core tap JSON: 24 Jun 01:10 UTC
Core cask tap JSON: 24 Jun 01:10 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: /usr/local/bin/bbedit --wait
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

```console
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
```

### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I upgraded the `msgpack` library from version 6.0.0 to version 6.0.1.

### What happened (include all command output)?

msgpack-c version 6.0.1 has a [bug](https://github.com/msgpack/msgpack-c/issues/1128) that causes the generated `msgpack-c.pc` pkg-config file to be invalid. The `libdir` and `includedir` variables are missing the prefix:
```
prefix=/opt/homebrew/Cellar/msgpack/6.0.1
exec_prefix=/opt/homebrew/Cellar/msgpack/6.0.1
libdir=lib
includedir=include

Name: MessagePack
Description: Binary-based efficient object serialization library
Version: 6.0.1
Libs: -L${libdir} -lmsgpack-c
Cflags: -I${includedir}
```

### What did you expect to happen?

msgpack-c version 6.0.0 generates a valid `msgpack-c.pc` file:
```
prefix=/opt/homebrew/Cellar/msgpack/6.0.0
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: MessagePack
Description: Binary-based efficient object serialization library
Version: 6.0.0
Libs: -L${libdir} -lmsgpack-c
Cflags: -I${includedir}
```

Notice that `libdir` and `includedir` correctly include the prefix.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
$ brew install msgpack
```

**Comment by SMillerDev** [comment]
This will automatically be fixed once upstream releases the next version so I'm closing this since there is nothing actionable on the brew side.

**Comment by fumoboy007** [comment]
Fixed by #175543.
